File: src/main/java/self/ed/main/DataDirs.java
package self.ed.main;

import java.nio.file.Path;
import java.nio.file.Paths;

import lombok.experimental.UtilityClass;

@UtilityClass
public class DataDirs {

  public static final Path DATA_DIR = Paths.get("data");
  public static final Path READY_DIR = DATA_DIR.resolve("ready");
  public static final Path REDUCER_FAILED_DIR = DATA_DIR.resolve("reducer-failed");
  public static final Path REDUCER_FIXED_DIR = DATA_DIR.resolve("reducer-fixed");
}

File: cpp-sudoku/src/data_dirs.cpp
#include "data_dirs.hpp"

namespace sudoku {

const std::filesystem::path DataDirs::DATA_DIR = "data";
const std::filesystem::path DataDirs::READY_DIR = DATA_DIR / "ready";
const std::filesystem::path DataDirs::REDUCER_FAILED_DIR = DATA_DIR / "reducer-failed";
const std::filesystem::path DataDirs::REDUCER_FIXED_DIR = DATA_DIR / "reducer-fixed";

} // namespace sudoku
